src/acvp_dsa.c

```diff
@@ -245,17 +249,19 @@ static ACVP_RESULT acvp_dsa_pqgver_init_tc (ACVP_CTX *ctx,
         stc->index = strtol((char *) index, NULL, 16);
     }
     if (seed) {
-        stc->seedlen = strlen((const char *)seed)/2;
-        rv = acvp_hexstr_to_bin((const unsigned char *) seed, stc->seed, ACVP_DSA_MAX_STRING);
+        rv = acvp_hexstr_to_bin((const unsigned char *) seed, stc->seed, ACVP_DSA_MAX_STRING, &(stc->seedlen));
         if (rv != ACVP_SUCCESS) {
             ACVP_LOG_ERR("Hex conversion failure (seed)");
             return rv;
         }
     }
     memcpy(stc->p, p, strlen((const char *)p));
     memcpy(stc->q, q, strlen((const char *)q));
+    stc->p_len = strlen((const char *)p);
```

Comment:
strnlen? General comment

Reply:
Looks like some of the dsa changes didn't make it into my commit... most of the 'strlen' offenses were in that file. Should be fixed with the update commit :)